ec/src/msm/variable_base.rs

```diff
@@ -28,6 +25,11 @@ impl VariableBaseMSM {
         let zero = G::Projective::zero();
         let window_starts: Vec<_> = (0..num_bits).step_by(c).collect();
 
+        let size = ark_std::cmp::min(bases.len(), scalars.len());
+        let scalars = &scalars[..size];
+        let bases = &bases[..size];
+        let scalars_and_bases = scalars.iter().zip(bases).filter(|(s, _)| !s.is_zero());
```

Comment:
This change is because in some downstream projects, (poly-commit) algorithms that use MSM are slower if the vectors are mismatched. The `zip` should fix this, but in benchmarks it doesn't.

(For that matter, neither does this new code, so I can remove it if that's better)

Reply:
Looks to me like this is due to line 19, where `c` is set just on the scalars length? That should probably be moved down, and calculated based on size.

Reply:
To clarify, this was happening when `bases.len() > scalars.len()`, so the `c` calculation wasn't affected.

(But you're right, we should move the `c` calculation down anyway)

Reply:
Also scalars should never be bigger then the number of bases, right?

Reply:
Hm not sure if that's necessary; you could imagine a case where zero-padding makes `scalars` larger, but because the extras are zero, it doesn't affect the result

(This might be happening in `ark-poly-commit` already)

Reply:
Hrmm, the function doesn't really make sense in the case where its non-zero. (And in the case where there are more 0's than bases, its being inefficient in allocating them)

Regardless of if this function accepts extra zeroes, imo we should (eventually) change poly-commit to not allocate extra zeros